external/webserver/http_string_util.c

```diff
@@ -87,7 +87,11 @@ int http_separate_header(const char *src, int *method, char *url, int *httpver)
 	url_start = divide[0] + 1;
 	url_length = divide[1] - url_start;
 
+	if(url_length < 0) {
```

Comment:
If url_length is less than zero, this functino has to return with failure, not going through the rest of the function. So this checkpoint may be not required. I recommend that you need to check the divide_pos to assure the url_length is greather than zero.